python/src/worker/tasks/gene_expression.py

```diff
@@ -26,50 +27,54 @@ def _format_request(self):
     def compute(self):
         request = self._format_request()
 
-        r = requests.post(
+        response = requests.post(
             f"{config.R_WORKER_URL}/v0/runExpression",
             headers={"content-type": "application/json"},
             data=json.dumps(request),
         )
-        # raise an exception if an HTTPError if one occurred because otherwise r.json() will fail
-        r.raise_for_status()
-        resultR = r.json()
-        truncatedR = resultR["truncatedExpression"]
-        resultR = resultR["rawExpression"]
+
+        # raise an exception if an HTTPError occurred
+        # as otherwise response.json() will fail
+        response.raise_for_status()
+        result = response.json()
+
+        error = result.get("error", False)
+        if error:
+            user_message = error.get("user_message", "")
+            err_code = error.get("error_code", "")
+            raise WorkerException(err_code, user_message)
+
+        data = result.get("data")
+        truncatedExpression = data["truncatedExpression"]
+        rawExpression = data["rawExpression"]
         result = {}
 
-        if not len(resultR):
-            result[genes[0]] = {
-                "error": 404,
-                "message": "Gene {} not found!".format(genes[0]),
-            }
+        for gene in rawExpression.keys():
 
-        else:
-            for gene in resultR.keys():
+            view = rawExpression[gene]
```

Comment:
what is a view? is there a more clear name?

Reply:
yea, it should contain the expression of a gene - not all the genes. I've changed the name.